tls/extensions/s2n_server_key_share.c

```diff
@@ -22,28 +22,120 @@
 #include "utils/s2n_safety.h"
 #include "tls/s2n_tls13.h"
 
+static int s2n_server_key_share_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_server_key_share_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_server_key_share_extension = {
+    .iana_value = TLS_EXTENSION_KEY_SHARE,
+    .is_response = false,
+    .send = s2n_server_key_share_send,
+    .recv = s2n_server_key_share_recv,
+    .should_send = s2n_extension_send_if_tls13_connection,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+int s2n_extensions_server_key_share_send_check(struct s2n_connection *conn);
+
+static int s2n_server_key_share_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+{
+    GUARD(s2n_extensions_server_key_share_send_check(conn));
+    notnull_check(conn);
+    notnull_check(out);
```

Comment:
you are checking conn not null after being used !

Reply:
We're just passing the conn pointer, not dereferencing it, so it can be NULL at that point without any issue.